dotnet/src/webdriver/Interactions/PointerInputDevice.cs

```diff
@@ -551,14 +512,14 @@ public override string ToString()
 
         private class PointerMoveInteraction : Interaction
         {
-            private IWebElement target;
-            private int x = 0;
-            private int y = 0;
+            private readonly IWebElement? target;
+            private readonly int x = 0;
+            private readonly int y = 0;
             private TimeSpan duration = TimeSpan.MinValue;
-            private CoordinateOrigin origin = CoordinateOrigin.Pointer;
-            private PointerEventProperties eventProperties;
+            private readonly CoordinateOrigin origin = CoordinateOrigin.Pointer;
+            private readonly PointerEventProperties eventProperties;
 
-            public PointerMoveInteraction(InputDevice sourceDevice, IWebElement target, CoordinateOrigin origin, int x, int y, TimeSpan duration, PointerEventProperties properties)
+            public PointerMoveInteraction(InputDevice sourceDevice, IWebElement? target, CoordinateOrigin origin, int x, int y, TimeSpan duration, PointerEventProperties properties)
```

Comment:
Comment for us to double-check whether `IWebElement? target` might be null.

Reply:
The challenge here is that `PointerMoveInteraction` is doing the work of two types, based on the `CoordinateOrigin` value. We have two entry points: one which has `CoordinateOrigin.Element`, and the other which forbids `CoordinateOrigin.Element`. 
![image](https://github.com/user-attachments/assets/924e63c0-8c23-48ab-85cb-e0b8c3eef18d)

For this reason, we know that if `origin is CoordinateOrigin.Element`, then `target is not null`. This can be split out into two different types for simplicity.

Reply:
The `PointerMoveInteraction` type is `private`, so we could change some logic around

Additionally, since this is just the `ToString()` method, we can double-check that `target` is not null, just to protect against any future refactorings.

Reply:
Oof, I just tried calling this: `CreatePointerMove(null, 0, 0, TimeSpan.FromSeconds(2)).ToString()` and got a null reference exception. This bug exists in the current code. Ideally this should be an `ArgumentException` I believe? You can't request to move from a null element?

Reply:
This is why I left initial comment to double check. Probably `target` null is not allowed by spec (remote end). Can you share internal logs? Ah no, probably it was not even went to remote end.

Reply:
Hm, is it only about `.ToString()`?

Reply:
The spec allows a few things to be `origin`: https://w3c.github.io/webdriver/#dfn-process-a-pointer-move-action

And the `ToDictionary()` method handles some logic depending on whether `target` is null:
![image](https://github.com/user-attachments/assets/e6599cd2-a75f-4efa-a986-df07b83ffdd8)

> Hm, is it only about `.ToString()`?

Yes, but I'm not sure whether this even works when sent to the server, since `origin` is `element`, which is not supported by the spec.

Reply:
It creates the following dictionary:
```json
{
  "type": "pointerMove",
  "duration": 2000,
  "origin": "element",
  "x": 0,
  "y": 0
}
```

and so it doesn't work:
![image](https://github.com/user-attachments/assets/06f87d13-ac0d-4573-9fc9-ec5820c3cea7)

I think it's safe to throw an `ArgumentNullException` for this overload when the element is `null`.

Reply:
`ToString()` is involved in json serialization?

Do you see simple ways to resolve it, or I should open IDE?

Reply:
No, this doesn't involve the `ToString()` method. This is about using the action like normal. The problem is, this is an invalid state for the interaction to be in, so the `ToString()` method doesn't guard against it. if `origin` is `Element`, then `target` cannot be null.

We can resolve it simply by adding this little argument check:
![image](https://github.com/user-attachments/assets/c0cdbf4c-ad26-4f14-a38e-90e3c0a21e21)

Reply:
I am pretty sure that `target, x, y` is a valid set of parameters. We should understand the expected logic.

Reply:
The `origin` property can be 3 things: string `"viewport"`, string `"pointer"`, or object
```json
{
  "element-6066-11e4-a52e-4f735466cecf": "f.8F0C47CA9FBF6735FEA36184E3852F05.d.C48927D06ABA43F7BB2B7EA08D439DBB.e.5"
}
```

If we specify we are using `Element`, then the target element cannot be null.

Reply:
> Am I right that this hidden knowledge is private? If yes, we can leave it as is.

This is a debacle of the `Actions` vs `ActionsBuilder` class. The `Actions` class exposes higher-level operations which can be built up. However, it does not expose a mechanism to append any `Interaction` to it's underlying `ActionBuilder`.

For that reason, it is not actually possible for users to hit this error. As such, I will remove this extra validation from this PR.

My proposal is to make `ActionsBuilder` into an `internal` class, and to make methods like `CreatePointerMove` into `internal` methods. It is not actually possible for users to use these methods, and operations such as

```csharp
var actions = new Actions(driver);

actions.GetActivePointer().CreatePointerMove(element, 0, 0, duration);
 ```

Is very confusing. It does not actually do anything, but users might think it would. Better to remove these methods from the user's eyes. 

This is completely separate from this PR, and I will open a separate issue. For now, I will remove any real changes from this PR, and focus on nullability.

Reply:
Opened issue #14873 for this concern